### PR TITLE
RMET-3046 H&F Cordova Plugin - Migrate RequestPermissions to HealthConnect

### PR DESCRIPTION
**Pull request by alexgerardojacinto**
## Description

- This PR re-implements the `RequestPermissions` feature - `initAndRequestPermissions` method in the Cordova bridge.
- It also updates the version of the `OSHealthFitnessLib-Android` library

## Context

We're migrating the H&F library from the Google Fit Android API to the Health Connect SDK.

References: https://outsystemsrd.atlassian.net/browse/RMET-3046

## Type of changes

- [ ] Fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [x] Android
- [ ] iOS
- [ ] JavaScript

## Tests

- Unit tests for the `RequestPermissions` feature were added to `src/test/kotlin/com/outsystems/plugins/healthfitness/RequestPermissionsTest.kt`.

- MABS 9 build: https://intranet.outsystems.net/MABS/BuildDetail.aspx?BuildId=94bf797e81f8658511d4c27a556a8bd06ae16352
- MABS 10 build: https://intranet.outsystems.net/MABS/BuildDetail.aspx?BuildId=017c0cbd5c4c3fc5ab855f4b01b19bd909b904d8

- Tested in Android 14 and Android 13.

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 The changes documented here do not include those from the original repository.
 
+## [Unreleased]
+
+## 2023-08-25
```
Comment (OS-ricardomoreirasilva) [inline CHANGELOG.md:11]: Bad copy paste?

File: src/android/com/outsystems/plugins/healthfitness/OSHealthFitness.kt [hunk 2]
```diff
@@ -248,27 +279,37 @@ class OSHealthFitness : CordovaImplementation() {
...
+            },
+            {
+                sendPluginResult(null, Pair(it.code.toString(), it.message))
+            }
+        )
     }
 
+
+
+
```
Comment (OS-ricardomoreirasilva) [inline src/android/com/outsystems/plugins/healthfitness/OSHealthFitness.kt:295]: You can remove these empty lines.

**Review by OS-ricardomoreirasilva — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (alexgerardojacinto) [reply CHANGELOG.md]: Woops, yes indeed. Fixed!

File: src/android/com/outsystems/plugins/healthfitness/OSHealthFitness.kt (hunk 2 above)
Comment (alexgerardojacinto) [reply src/android/com/outsystems/plugins/healthfitness/OSHealthFitness.kt]: Fixed!

**Pull request merged**